framework/integration/test/basic/basic_test.go

```diff
@@ -48,50 +56,60 @@ func Test_Finalizer_Integration_Basic(t *testing.T) {
 		},
 		Data: map[string]string{},
 	}
-	// We create a configmap which does not have any finalizers.
-	createdConfigMap, err := client.CreateConfigMap(cm)
+	// We create an object which does not have any finalizers.
+	createdObj, err := testClient.CreateObject(testNamespace, cm)
 	if err != nil {
 		t.Fatal("expected", nil, "got", err)
 	}
 
-	// We directly pass the configmap to UpdateFunc.
-	operatorkitFramework.UpdateFunc(createdConfigMap, createdConfigMap)
+	// We directly pass the object to UpdateFunc.
+	operatorkitFramework.UpdateFunc(createdObj, createdObj)
 
-	resultConfigMap, err := client.GetConfigMap(configMapName, testNamespace)
+	resultObj, err := testClient.GetObject(configMapName, testNamespace)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	resultObjAccessor, err := meta.Accessor(resultObj)
 	if err != nil {
 		t.Fatal("expected", nil, "got", err)
 	}
 
 	// We verify that the correct finalizer has been set during UpdateFunc.
-	if !reflect.DeepEqual(resultConfigMap.GetFinalizers(), expectedFinalizers) {
-		t.Fatalf("finalizers == %v, want %v", resultConfigMap.GetFinalizers(), expectedFinalizers)
+	if !reflect.DeepEqual(resultObjAccessor.GetFinalizers(), expectedFinalizers) {
+		t.Fatalf("finalizers == %v, want %v", resultObjAccessor.GetFinalizers(), expectedFinalizers)
 	}
 
-	// We delete out configmap.
-	err = client.DeleteConfigMap(configMapName, testNamespace)
+	// We delete our object.
+	err = testClient.DeleteObject(configMapName, testNamespace)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+	resultObj, err = testClient.GetObject(configMapName, testNamespace)
 	if err != nil {
 		t.Fatal("expected", nil, "got", err)
 	}
-	resultConfigMap, err = client.GetConfigMap(configMapName, testNamespace)
+
+	resultObjAccessor, err = meta.Accessor(resultObj)
```

Comment:
This is the main reason why this works, all relevant data for the test is available from the accessor. There is no need to really care about the rest of the object for the testcase.